### PR TITLE
add listObjects

### PR DESCRIPTION
**Pull request by AWitcherONS**
### What

add listObjects that will list all files for a given bucket for dp-cpih-export policy test

### How to review

code coplies with current standards for libraries
make build still works
make test sill passes

### Who can review
anyone but me

Describe who worked on the changes, so that other people can review.

### CODE REVIEW

File: get.go [hunk 1]
```diff
@@ -202,3 +202,24 @@ func (cli *Client) PutBucketPolicy(BucketName string, policy string) (*s3.PutBuc
...
+
+	result, err := cli.sdkClient.ListObjects(&s3.ListObjectsInput{
+		Bucket: aws.String(BucketName),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "NotFound": // s3.ErrCodeNoSuchKey does not work, aws is missing this error code so we hardwire a string
+				return nil, nil
```
Comment (lindenmckenzie) [inline get.go]: Not sure if this matches with the test case? 

Test case seems to imply we should be returning an error, but this code seems explicitly wired up to return nil.

File: go.mod [hunk 2]
```diff
@@ -9,22 +9,30 @@ retract (
...
 	github.com/gopherjs/gopherjs v1.17.2 // indirect
 	github.com/hokaccha/go-prettyjson v0.0.0-20211117102719-0474bc63780f // indirect
 	github.com/jmespath/go-jmespath v0.4.0 // indirect
 	github.com/jtolds/gls v4.20.0+incompatible // indirect
 	github.com/mattn/go-colorable v0.1.13 // indirect
-	github.com/mattn/go-isatty v0.0.18 // indirect
-	github.com/smartystreets/assertions v1.13.1 // indirect
-	golang.org/x/sys v0.7.0 // indirect
-	golang.org/x/text v0.8.0 // indirect
+	github.com/mattn/go-isatty v0.0.20 // indirect
```
Comment (lindenmckenzie) [inline go.mod:36]: Any reason to keep these as two separate require blocks?

**Review by lindenmckenzie — COMMENTED** [review]
Minor test comment.

File: go.mod (hunk 2 above)
Comment (AWitcherONS) [reply go.mod:36]: Thats how go mod tidy did it

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**